src/notebookMiddlewareAddon.ts

```diff
@@ -79,14 +81,49 @@ export class NotebookMiddlewareAddon implements Middleware, Disposable {
         notebookApi: IVSCodeNotebook,
         private readonly getClient: () => LanguageClient | undefined,
         private readonly traceInfo: (...args: any[]) => void,
-        fs: IFileSystem,
         cellSelector: string,
-        notebookFileRegex: RegExp
+        notebookFileRegex: RegExp,
+        private readonly pythonPath?: string
     ) {
-        this.converter = new NotebookConverter(notebookApi, fs, cellSelector, notebookFileRegex);
+        this.converter = new NotebookConverter(notebookApi, cellSelector, notebookFileRegex);
         this.didChangeCellsDisposable = this.converter.onDidChangeCells(this.onDidChangeCells.bind(this));
+
+        // Make sure a bunch of functions are bound to this. VS code can call them without a this context
+        this.handleDiagnostics = this.handleDiagnostics.bind(this);
+        this.didOpen = this.didOpen.bind(this);
+        this.didSave = this.didSave.bind(this);
+        this.didChange = this.didChange.bind(this);
+        this.didClose = this.didClose.bind(this);
+        this.willSave = this.willSave.bind(this);
+        this.willSaveWaitUntil = this.willSaveWaitUntil.bind(this);
     }
 
+    public workspace = {
```

Comment:
Adding this allows the middleware addon to be used as middleware by itself. 

This code here is basically the same thing that python does in its middleware, although it uses the active interpreter.

For the test code, I used CI_PYTHON_PATH variable and pass it into the instance of the middleware. 

When I implement this for actual notebooks, the ctor for the middleware will get the path to the interpreter.